cordapp-example/contracts-kotlin/src/main/kotlin/com/example/contract/CashContract.kt

```diff
@@ -0,0 +1,38 @@
+package com.example.contract
+
+import com.example.state.CashState
+import net.corda.core.contracts.CommandData
+import net.corda.core.contracts.Contract
+import net.corda.core.contracts.requireSingleCommand
+import net.corda.core.contracts.requireThat
+import net.corda.core.transactions.LedgerTransaction
+
```

Comment:
@xavierlepretre in my comments on the first revision of this contract I also pointed out to many validation rules; for instance the sum of input should be equal to the sum of output for the `move` command; also I don't see a `redeem` command.

Reply:
I found my comments: https://github.com/DZariusz/cordapp-template-kotlin/pull/1

@xavierlepretre did you advise against those remarks, or they were simply ignored?

Reply:
Sorry, just trying to be up to date; since I don't always follow the conversation between the 2 of you 🙂

Reply:
I think he simply copy pasted.

Reply:
Ok 🙂

Reply:
I think this is comment for module 4, but this repo is for module 3. so I close it.
And yes, I done a lot of copy-paste because it was easier to copy and adjust, when I had zero knowledge about corda :)